test/sqllogictest/joins.slt

```diff
@@ -602,3 +602,92 @@ CREATE TABLE join_fail (la date);
 
 query error NATURAL/USING join column "la" cannot be cast to uniform type: integer vs date
 SELECT la FROM l JOIN join_fail USING (la)
+
+# test that joins properly handle null keys
+statement ok
+CREATE TABLE l3 (la int, lb text)
```

Comment:
This query:

```materialize=> explain SELECT *  FROM t2  AS a1  LEFT  JOIN t2  AS a2  ON ( a1 .f2  = a2 .f1  )  WHERE   a1.f2  IS  NOT  NULL  AND a2 .f1  IS  NULL   ;
 Optimized Plan 
----------------
 %0 =          +
 | Constant    +
```

Is incorrectly judged to return no rows. With an outer join the ```WHERE``` clause can be satisfied even though propagating the ```IS [NOT] NULL``` through the equality in the ```ON``` clause may suggest that the predicate is impossible to satisfy.